gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/UserServiceImpl.java

```diff
@@ -199,9 +201,22 @@ public UserEntity connect(String userId) {
                     defaultApp.setSettings(settings);
 
                     try {
-                        applicationService.create(defaultApp, userId, true);
+                        List<EnvironmentEntity> environments = environmentService.findByUser(userId);
+                        for (EnvironmentEntity environment : environments) {
+                            Optional<ApplicationListItem> existingDefaultApp = applicationService.findAllByEnvironment(environment.getId())
+                                    .stream()
+                                    .filter(app -> defaultApp.getName().equals(app.getName())
+                                            && defaultApp.getDescription().equals(app.getDescription())
+                                            && ApplicationType.SIMPLE.name().equals(app.getType()))
+                                    .findFirst();
+
+                            if (!existingDefaultApp.isPresent()) {
+                                applicationService.create(defaultApp, userId, environment.getId());
+                            }
+                        }
```

Comment:
It won't work since multiple "Default applications" with the same name/description/type can exist in an environment, but one per user.
Here, you are in a "first login" context, so no default application exists for the current user yet.

```suggestion
                        environmentService.findByUser(userId).stream()
                            .map(EnvironmentEntity::getId)
                            .forEach(environmentId -> applicationService.create(defaultApp, userId, environmentId));
                        }
```